src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -776,6 +777,29 @@ public HttpRequestInterceptionBuilder HavingPriority(int? priority)
         public HttpRequestInterceptionBuilder ForRequestHeader(string name, string value)
             => ForRequestHeader(name, new[] { value });
 
+        /// <summary>
+        /// Sets an Authentication header to intercept a request for.
+        /// </summary>
+        /// <param name="authenticationHeader">The authentication credential of the request.</param>
+        /// <returns>
+        /// The current <see cref="HttpRequestInterceptionBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="authenticationHeader"/> is <see langword="null"/>.
+        /// </exception>
+        /// <remarks>
+        /// HTTP request authentication are only tested for interception if the URI requested was registered for interception.
+        /// </remarks>
+        public HttpRequestInterceptionBuilder ForRequestAuthentication(AuthenticationHeaderValue authenticationHeader)
```

Comment:
I'm not convinced of the need to add this to the library.

It is possible to do with `ForRequestHeader()` already, and ties the public surface of the library more tightly to the implementation details of .NET itself (i.e. the implementation of `AuthenticationHeaderValue` to strongly type the value).